fedora-builder/src/main/java/org/dataconservancy/nihms/builder/fs/DepositFileLabelMaker.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 Johns Hopkins University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataconservancy.nihms.builder.fs;
+
+import org.dataconservancy.nihms.model.DepositFileType;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A utility class for generating unique labels for manifest entries for {@code DepositFile}s.
+ *
+ * @author jrm@jhu.edu
+ */
+
+public class DepositFileLabelMaker {
+
+    //these are the types for which we will supply labels
+    private static final DepositFileType[] labeledTypes = new DepositFileType[] {
+            DepositFileType.figure,
+            DepositFileType.supplemental,
+            DepositFileType.table
+    };
+
+    private static final Set<DepositFileType> types = new HashSet<>(Arrays.asList(labeledTypes));
+
+    private static Map<DepositFileType, Set<String>> usedFileLabels = createLabelMap();
+
+    private static  Map<DepositFileType, Set<String>> createLabelMap() {
+        Map<DepositFileType, Set<String>> labelMap = new HashMap<>();
+        for (DepositFileType fileType : types) {
+            labelMap.put(fileType, new HashSet<>());
+        }
+        return labelMap;
+    }
+
+    /**
+     * Return a unique label for a {@code }DepositFile}. If the type of file is not to be labeled, return an empty string.
+     *
```

Comment:
@jrmartino can we document which types are expected to be labeled, and which are not?  If this is due to a spec, can a reference to the spec be cited (e.g. "NIH Bulk Submission Specification dated xx/yy/zz")

Reply:
@jrmartino sorry, to be specific: please update the javadoc to reflect which types are expected to be labeled, and a citation for the spec!  Thanks.